### PR TITLE
Add EOS scan

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: src/main/treeDataProviders/utils/utils.ts [hunk 1]
```diff
@@ -14,6 +14,14 @@ export class Utils {
         return path.substring(path.lastIndexOf(pathUtils.sep) + 1);
     }
 
+    public static tryRelativePath(full: string, potentialParent?: string): string {
+        if (potentialParent && full.startsWith(potentialParent)) {
+            let localPath: string = full.substring(potentialParent.length + 1);
+            return './' + localPath;
```
Comment (yahavi) [inline src/main/treeDataProviders/utils/utils.ts:20]: Can we change it to this?
```suggestion
            return full.substring(potentialParent.length + 1);
```

Explanation - I'm not sure that "./" would work on Windows. Do we really need it?

File: src/main/types/issuesData.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class WorkspaceIssuesData {
      * @returns true if at least one issue exists
      */
     public hasIssues(): boolean {
-        return this.descriptorsIssuesData.length > 0;
+        return this.descriptorsIssuesData.length > 0 || (this.eosScan && this.eosScan.filesWithIssues && this.eosScan.filesWithIssues.length > 0);
```
Comment (yahavi) [inline src/main/types/issuesData.ts:24]: A bit shorter
```suggestion
        return this.descriptorsIssuesData.length > 0 || this.eosScan?.filesWithIssues?.length > 0;
```

File: src/main/webviews/detailsWebView.ts [hunk 3]
```diff
@@ -10,39 +10,41 @@ import { LogManager } from '../log/logManager';
...
-                prevActiveTreeNode = page;
-            })
+            vscode.commands.registerCommand('jfrog.view.details.page', (page: IDependencyPage | IEosPage) => this.updateWebview(page, context))
         );
     }
+
+    public updateWebview(data: any, context: vscode.ExtensionContext) {
+        if (!this._panel) {
+            this._panel = createWebview(context);
+            this._panel.onDidChangeViewState(e => {
```
Comment (yahavi) [inline src/main/webviews/detailsWebView.ts:25]: `e` is not enough informative
```suggestion
            this._panel.onDidChangeViewState((webviewChanged: vscode.WebviewPanelOnDidChangeViewStateEvent) => {
```

File: src/main/webviews/detailsWebView.ts [hunk 4]
```diff
@@ -10,39 +10,41 @@ import { LogManager } from '../log/logManager';
...
-                    this._logManager.logMessage('Opening webview page with data:\n' + JSON.stringify(page), 'DEBUG');
-                    updateWebview(panel, page);
-                }
-                prevActiveTreeNode = page;
-            })
+            vscode.commands.registerCommand('jfrog.view.details.page', (page: IDependencyPage | IEosPage) => this.updateWebview(page, context))
         );
     }
+
+    public updateWebview(data: any, context: vscode.ExtensionContext) {
```
Comment (yahavi) [inline src/main/webviews/detailsWebView.ts:22]: Let's add a description above this method

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/main/treeDataProviders/utils/utils.ts (hunk 1 above)
Comment (attiasas) [reply src/main/treeDataProviders/utils/utils.ts:20]: removed the translation to relative.

File: src/main/treeDataProviders/utils/utils.ts [hunk 5]
```diff
@@ -14,6 +14,14 @@ export class Utils {
         return path.substring(path.lastIndexOf(pathUtils.sep) + 1);
     }
 
+    public static tryRelativePath(full: string, potentialParent?: string): string {
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/treeDataProviders/utils/utils.ts:17]: removed

File: src/main/scanLogic/scanRunners/eosScan.ts [hunk 6]
```diff
@@ -67,49 +82,118 @@ export class EosRunner extends BinaryRunner {
...
+                            analyzeIssue,
+                            rulesFullDescription.get(analyzeIssue.ruleId)
+                        );
+                        let issueLocation: EosIssueLocation = this.getOrCreateIssueLocation(fileIssue, location.physicalLocation);
+                        if (analyzeIssue.codeFlows) {
+                            // Check if exists flows for the current location in this issue
+                            for (const codeFlow of analyzeIssue.codeFlows) {
+                                for (const threadFlow of codeFlow.threadFlows) {
+                                    // The last location in the threadFlow should match the location of the issue
+                                    let potential: AnalyzeLocation = threadFlow.locations[threadFlow.locations.length - 1].location;
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/main/scanLogic/scanRunners/eosScan.ts:125]: always location, not always codeFlow and there are locations with multiple codeFlows for them

**Pull request merged**